File: apps/elixir/lib/fizz_buzz.ex
defmodule FizzBuzz do
  def generate(n) do
    cond do
      rem(n, 15) == 0 -> "FizzBuzz"
      rem(n, 3) == 0  -> "Fizz"
      rem(n, 5) == 0  -> "Buzz"
      true            -> Integer.to_string(n)
    end
  end

  def generate_list do
    1..100
    |> Enum.map(&generate/1)
  end
end

File: apps/haskell/src/FizzBuzz.hs
module FizzBuzz
  ( compose
  , generate
  , filterList
  , generateWith
  , generateList
  , transform
  ) where

generate :: Int -> String
generate n
  | n `mod` 15 == 0 = "FizzBuzz"
  | n `mod` 3 == 0  = "Fizz"
  | n `mod` 5 == 0  = "Buzz"
  | otherwise       = show n

generateWith :: (Int -> String) -> Int -> String
generateWith rule = rule

compose :: (b -> c) -> (a -> b) -> a -> c
compose = (.)

generateList :: Int -> [String]
generateList n = map generate [1 .. n]

transform :: (a -> b) -> [a] -> [b]
transform = map

filterList :: (a -> Bool) -> [a] -> [a]
filterList = filter
